source/common/thread_safe_monotonic_time_stopwatch.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include "nighthawk/common/stopwatch.h"
+
+#include "external/envoy/source/common/common/lock_guard.h"
+#include "external/envoy/source/common/common/thread.h"
+
+namespace Nighthawk {
+
+/**
+ * Utility class for thread safe tracking of elapsed monotonic time.
+ * Example usage:
+ *
+ * ThreadSafeMontonicTimeStopwatch stopwatch;
+ * int i = 0;
+ * do {
+ *   std::cerr << stopwatch.getElapsedNsAndReset() <<
+ *    "ns elapsed since last iteration." << std::endl;
+ * } while (++i < 100);
+ */
+class ThreadSafeMontonicTimeStopwatch : public Stopwatch {
+public:
+  /**
+   * Construct a new ThreadSafe & MontonicTime-based Stopwatch object.
+   */
+  ThreadSafeMontonicTimeStopwatch() : start_(Envoy::MonotonicTime::min()) {}
+
+  /**
+   * @param time_source used to obtain a sample of the current monotonic time.
+   * @return uint64_t 0 on the first invocation, and the number of elapsed nanoseconds since the
```

Comment:
Well, while I didn't consider `absl::Duration` (I didn't know it existed). But I did consider `std::chrono::duration`, and ended up leaning towards `uint64_t` because that:
- can't turn negative, which doesn't make sense here
- is less prone to some of the trickyness with conversions vs the underlying types I ran into over at https://github.com/envoyproxy/envoy/pull/9229#issuecomment-562498924. 

Having said that, I feel this is a subjective matter, so I'd be happy to change this if you lean towards `absl::Duration` after reading my considerations above.

Reply:
That makes sense.  I don't know if it's worth bringing in absl::Duration just for this.  If we wanted to, we could try to bring in absl::Duration on a more global change later.